File: go-api/models/blocklist.go
package models

import (
	"github.com/google/uuid"
)

type BlockList struct {
	Blocker uuid.UUID `gorm:"column:userid1;type:uuid;primary_key"`
	Blocked uuid.UUID `gorm:"column:userid2;type:uuid;primary_key"`
}

func (BlockList) TableName() string {
	return "chat_data.user_blocklist"
}

File: python-api/models/blocklist.py
import uuid

from sqlalchemy.dialects.postgresql import UUID

from db import db, ma


class BlocklistModel(db.Model):
    __table_args__ = { "schema": "chat_data" }
    __tablename__ = "user_blocklist"

    blocker_id = db.Column(
        "userid1",
        UUID(as_uuid=True), nullable=False, primary_key=True
    )
    blocked_id = db.Column(
        "userid2",
        UUID(as_uuid=True), nullable=False, primary_key=True
    )
    created_time = db.DateTime(timezone=True)


class BlocklistSchema(ma.SQLAlchemySchema):
    class Meta:
        model = BlocklistModel

    blocker_id = ma.auto_field()
    blocked_id = ma.auto_field()
    created_time = ma.auto_field()
